dspace/config/registries/dspace-types.xml

```diff
@@ -16,5 +16,18 @@
         <scope_note></scope_note>
     </dc-type>
 
+    <dc-type>
+        <schema>dspace</schema>
+        <element>agreements</element>
+        <qualifier>end-user</qualifier>
```

Comment:
I would prefer to have `dspace.agreements.acceptedDate` storing the timestamp of the acceptance. Anyway it would be better to add some scope note about how it will be used

Reply:
@abollini : I think you might be misunderstanding the point of this metadata field.  There's not just a *single* user/usage agreement, so the field `dspace.agreements.acceptedDate` seems wrong, as it implies there's one agreement with one date of acceptance.

Instead, there's (at least) two different agreements required by the GDPR Working Group:
1. A cookie consent form, described in more detail here: https://github.com/DSpace/dspace-angular/issues/735#issuecomment-670214947
2. An End User Agreement, described in more detail here: https://github.com/DSpace/DSpace/issues/2808#issuecomment-670215129

So, this PR is creating two new fields:
1. `dspace.agreements.cookies` to store an EPerson's answer to the Cookie consent form. My assumption is this field will store the list of cookies that the EPerson has chosen to allow (opt-in approach).  It likely cannot be just a true/false field as we had talked previously about using something like https://klaro.kiprotect.com/#features
2. `dspace.agreements.end-user` to store an EPerson's answer to the End User agreement.  My assumption is that this field _may be_ simply `true` or `false` based on whether they agree with the End User agreement.  However, we could choose to have this field store the date/time of agreement.
    1. Arguably, if this field is ever `false` or `null` then that EPerson account should be blocked until they agree.  It should not be possible to use DSpace if you don't agree to the End User Agreement.

Overall, I agree that we need to add in `<scope_note>` descriptions to both of these fields in order to describe the usage and valid/accepted values.  Currently, while the fields look OK at a glance, I've had to make assumptions on how I expect them to be used (but I'm not sure whether my assumptions are correct).

@pnbecker : You may also want to review the direction of this small PR to provide any feedback on behalf of the GDPR Working Group

Reply:
>1. `dspace.agreements.cookies` to store an EPerson's answer to the Cookie consent form. My assumption is this field will store the list of cookies that the EPerson has chosen to allow (opt-in approach).  It likely cannot be just a true/false field as we had talked previously about using something like https://klaro.kiprotect.com/#features

That is correct, it will store a copy of the klaro cookie, which is an array of cookieName/boolean pairs. The contents of this metadata field wil be set as the klaro cookie for the session every time you log in, and kept up to date every time you change something in the klaro panel.

>2. `dspace.agreements.end-user` to store an EPerson's answer to the End User agreement.  My assumption is that this field _may be_ simply `true` or `false` based on whether they agree with the End User agreement.  However, we could choose to have this field store the date/time of agreement.
    i. Arguably, if this field is ever `false` or `null` then that EPerson account should be blocked until they agree.  It should not be possible to use DSpace if you don't agree to the End User Agreement.

Correct. If this metadata field has any value other than true, the user will see the end-user agreement immediately after logging in, and can't do anything else until they agree. Disagreeing or canceling will log them out again. If there is an updated version of the user agreement the idea is that you delete this field for all EPersons, so they all have to agree again. We could store a timestamp here instead if that's preferred but I don't see a use case for it at the moment.